packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsList.vue

```diff
@@ -23,177 +23,111 @@
-			type: Boolean,
-			default: false,
-		},
-		workflow: {
-			type: Object as PropType<IWorkflowDb>,
-			required: true,
-		},
-		executions: {
-			type: Array as PropType<ExecutionSummary[]>,
-			default: () => [],
-		},
-		filters: {
-			type: Object as PropType<ExecutionFilterType>,
-			default: () => ({}),
-		},
-		execution: {
-			type: Object as PropType<ExecutionSummary> | null,
-			default: null,
-		},
-		loadingMore: {
-			type: Boolean,
-			default: false,
-		},
-	},
-	emits: [
-		'execution:delete',
-		'execution:stop',
-		'execution:retry',
-		'update:auto-refresh',
-		'update:filters',
-		'load-more',
-		'reload',
-	],
-	setup() {
-		const externalHooks = useExternalHooks();
-		const router = useRouter();
-		const workflowHelpers = useWorkflowHelpers({ router });
-		const { callDebounced } = useDebounce();
-
-		return {
-			externalHooks,
-			workflowHelpers,
-			callDebounced,
-			...useToast(),
-			...useMessage(),
-		};
-	},
-	computed: {
-		...mapStores(useTagsStore, useNodeTypesStore, useSettingsStore, useUIStore, useNpsSurveyStore),
-		temporaryExecution(): ExecutionSummary | undefined {
-			const isTemporary = !this.executions.find((execution) => execution.id === this.execution?.id);
-			return isTemporary ? this.execution : undefined;
-		},
-		hidePreview(): boolean {
-			return this.loading || (!this.execution && this.executions.length);
-		},
-		filterApplied(): boolean {
-			return this.filters.status !== 'all';
-		},
-		workflowDataNotLoaded(): boolean {
-			return this.workflow.id === PLACEHOLDER_EMPTY_WORKFLOW_ID && this.workflow.name === '';
-		},
-		requestFilter(): IDataObject {
-			return executionFilterToQueryFilter({
-				...this.filters,
-				workflowId: this.workflow.id,
-			});
-		},
-	},
-	watch: {
-		execution(value: ExecutionSummary) {
-			if (!value) {
-				return;
-			}
+);
 
-			this.$router
-				.push({
-					name: VIEWS.EXECUTION_PREVIEW,
-					params: { name: this.workflow.id, executionId: value.id },
-				})
-				.catch(() => {});
-		},
-	},
-	methods: {
-		async onDeleteCurrentExecution(): Promise<void> {
-			this.$emit('execution:delete', this.execution?.id);
-		},
-		async onStopExecution(): Promise<void> {
-			this.$emit('execution:stop', this.execution?.id);
-		},
-		async onRetryExecution(payload: { execution: ExecutionSummary; command: string }) {
-			const loadWorkflow = payload.command === 'current-workflow';
+onBeforeRouteLeave(async (to, _, next) => {
+	if (getNodeViewTab(to) === MAIN_HEADER_TABS.WORKFLOW) {
+		next();
+		return;
+	}
 
-			this.$emit('execution:retry', {
-				id: payload.execution.id,
-				loadWorkflow,
-			});
+	await workflowHelpers.promptSaveUnsavedWorkflowChanges(next, {
+		async confirm() {
+			const saved = await workflowHelpers.saveCurrentWorkflow({}, false);
+			if (saved) {
+				await npsSurveyStore.fetchPromptsData();
+			}
+			uiStore.stateIsDirty = false;
+			return true;
 		},
-	},
+		async cancel() {
+			uiStore.stateIsDirty = false;
+		},
+	});
```

Comment:
There's no need to provide `confirm` and `cancel`. What you wrote here is the default behaviour in `promptSaveUnsavedWorkflowChanges`.